### ISSUE TITLE
Specifying small-dark logo for login screen doesn't work

### ISSUE THREAD

**Issue by Johnny2Shoes** [roundcube/roundcubemail#8298]
Roundcube v1.5 - Elastic skin

Hi, I'm trying to configure the new dark mode with our logos, unfortunately our company logo is a dark blue that doesn't work with a dark background. [I noticed that a new option small-dark](https://github.com/roundcube/roundcubemail/blob/035c325d5767757fc0bba35ae4ea30655a7a92f9/config/defaults.inc.php#L499) was added to the config options that are overrideable in the config.inc.php, however in my situation this appears to being ignored. 

Here is my skin_logo section

`$config['skin_logo'] = array(
    "elastic:*" => "acmecorp/acmecorp.svg",
    "elastic:login" => "acmecorp/blue.svg",
    "elastic:login[small]" => "acmecorp/blue.svg",
    "elastic:login[small-dark]" => "acmecorp/white_small.svg",
    "elastic:*[favicon]" => "acmecorp/favicon.ico",
    "elastic:login[favicon]" => "acmecorp/favicon.ico",
    "elastic:*[small]" => "acmecorp/acmecorp_white.svg",
    "login" => "acmecorp/blue.svg",
    "login[dark]" => "acmecorp/white_small.svg",
    "login[small-dark]" => "acmecorp/white_small.svg",
    "[print]" => "acmecorp/blue.svg",
    "[favicon]" => "acmecorp/favicon.ico"
  );`
  
On desktop mode I see the white_small.svg logo as expected, but on mobile I see blue.svg. 
 
Is this an issue in my configuration or Roundcube?

**Comment by johndoh** [comment]
I think I might have put the first 2 conditions the wrong way round here https://github.com/roundcube/roundcubemail/blob/master/skins/elastic/ui.js#L1733

I suggest changing it to small-dark > small > dark > normal

**Comment by Johnny2Shoes** [comment]
I've tested this on our pre-release server and it does indeed work. 

https://github.com/roundcube/roundcubemail/pull/8300

**Comment by vaelu** [comment]
I have a similar issue here with the `"customSkinName:*[favicon]"` option here (or the docs are not up-to-date).

In this case I want to set the same favicon for the whole skin (via config, not via the skin's meta.json).

Expected code:
```
$config['skin_logo'] = array(
  "customSkinName:login" => "https://example.com/image",
  "customSkinName:login[small]" => "https://example.com/image",
  "customSkinName:*[favicon]" => "https://example.com/image",
  "customSkinName:[print]" => null,
);
```
But, this does not work. It shows the favicon in the webmail and on print, but on the login page, it takes the logo from `customSkinName:login`.

I've fixed it with:
```
$config['skin_logo'] = array(
  "customSkinName:login" => "https://example.com/image",
  "customSkinName:login[small]" => "https://example.com/image",
  "customSkinName:[favicon]" => "https://example.com/image",
  "customSkinName:login[favicon]" => "https://example.com/image",
  "customSkinName:[print]" => null,
);
```

Is this the expected behavior? And would it be possible to release some better explanation on the `skin_logo` config option?

**Comment by johndoh** [comment]
@vadu this is not related to the original issue. Please open a separates ticket for new issues.

I cannot recreate the behavior you describe with Roundcube 1.5 and the Elastic skin. I think you are talking about the same thing as #7456 and #7470 which was fixed in 4c8ea71e3a0dd9bebdf8002e973fd33fd6d66a83. It could also be an issue with your custom skin rather than the core.

**Comment by alecpl** [comment]
Fixed.
